mitosheet/mitosheet/api/get_search_matches.py

```diff
@@ -0,0 +1,44 @@
+#!/usr/bin/env python
+# coding: utf-8
+
+# Copyright (c) Saga Inc.
+# Distributed under the terms of the GPL License.
+import re
+import pandas as pd
+from typing import Any, Dict
+from mitosheet.types import StepsManagerType
+
+
+def get_search_matches(params: Dict[str, Any], steps_manager: StepsManagerType) -> str:
+    """
+    Finds the number of matches to a given search value in the dataframe.
+    """
+    sheet_index = params['sheet_index']
+    search_value = params['search_value']
+    df = steps_manager.dfs[sheet_index]
+
+    # First, generate a count for each value in the dataframe:
+    unique_value_counts = df.value_counts()
+
+    # Use the same regex for all searching
+    search_regex = re.compile(search_value, re.IGNORECASE)
+
+    # Then, filter for the search value:
+    matches = unique_value_counts.filter(regex=search_regex, axis=0)
+
+    # Then, sum the matches:
+    total_number_matches = matches.sum()
+
+    # Then, add the column names to the matches:
+    total_number_matches += len([col for col in df.columns if re.search(search_regex,str(col)) is not None])
+
+    # Find the indices of cells containing the search value
+    # Only search the first 1500 rows because the editor only shows the first 1500 rows. 
+    cell_matches = [{'row': i, 'col': j} for i in range(min(1500,len(df.index))) for j in range(len(df.columns)) if (re.search(search_regex,str(df.iloc[i, j])) is not None)]
```

Comment:
I think we should call these `rowIndex` and `colIndex` to be consistent with the rest of our codebase. Otherwise, it's not clear if it's the header or the index or what.

Reply:
I think `.iat` might be faster -- see here: https://pandas.pydata.org/docs/reference/api/pandas.DataFrame.iat.html#pandas.DataFrame.iat

also, if you save the num_rows and num_columns you can avoid the len calculations running every loop. For large datasets, this is a significant amount of computation - so let's break them out of the list comprehension and save them above